### ISSUE TITLE
Is there a way to use a forge mod instead of a plugin?

### ISSUE THREAD

**Issue by Telokis** [yariplus/nodebb-plugin-minecraft-integration#104]
Hello,

I'd really like to use your nodebb plugin but I can't use bukkit/cauldron/thermos/anything that allows both mods and plugins, I have to stick on forge only.
Is there a way to make it work even though I can't use plugins?

**Comment by yariplus** [comment]
Sure thing!

The "plugin" is universal, so it should work in the forge mods folder as well. Although, I have not updated it in quite a while. What version of forge are you using?

**Comment by Telokis** [comment]
Thanks for the quick answer!
I'm using the forge version 1614 for 1.7.10 (rather old) so I think it'll be fine!

**Comment by Telokis** [comment]
`[Server thread/INFO] [FML]: FML has found a non-mod file nodebb-integration-universal-0.8.0-alpha.10.jar in your mods directory. It will now be injected into your classpath. This could severe stability issues, it should be removed if possible.`
Is that intentional?

**Comment by yariplus** [comment]
Hmm, definitely not. I'll see what I can do.

**Comment by Telokis** [comment]
Thanks a lot!
By the way there were a donation button somewhere, can I have it back, please?

**Comment by Telokis** [comment]
Nevermind, I found it!

**Comment by yariplus** [comment]
Thank you kindly. I'll get this updated asap. Looks like I got it working with Thermos/KCauldron, but never finished vanilla forge yet.

**Comment by Telokis** [comment]
Yeah, I browsed the code a bit and you seem to have only put the skeleton right there. If you want I'm also a programmer, if you take a little bit of time to tell me where to start, I'd probably be able to help you.

**Comment by yariplus** [comment]
Sure! It would be mostly busy work. Copy the file structure from the bukkit folder, rename the files, and change the functions to call forge methods instead of bukkit ones.

It already compiles and loads in mc, just remove any commented out forge sections from build.gradle in the universal section, and run `gradle package`.

One problem is the shadow plugin copies the forge libraries to the completed jar. I need to add all of the forge libraries to the exclusion list in forge's build.gradle. Not sure how to do that without naming all the libraries individually.

**Comment by Telokis** [comment]
Sorry but I think I won't have enough spare time to help you.
I work full time and I have a youtube channel to manage when I come back after work.

**Comment by yariplus** [comment]
No problem! I'm about in the same boat, but I'll manage it!

**Comment by yariplus** [comment]
Okay, I think I got it building correctly now. Won't be much longer.

**Comment by Telokis** [comment]
Alright I'm glad to hear that! Good luck!
